zookeeper-server/src/test/java/org/apache/zookeeper/server/admin/JettyAdminServerTest.java

```diff
@@ -234,6 +236,35 @@ public void testQuorum() throws Exception {
                 "waiting for server 2 down");
     }
 
+    @Test
+    public void testForceHttpsPortUnificationEnabled() throws Exception {
+        testForceHttps(true);
+    }
+
+    @Test
+    public void testForceHttpsPortUnificationDisabled() throws Exception {
+        testForceHttps(false);
+    }
+
+    private void testForceHttps(boolean portUnification) throws Exception {
+        System.setProperty("zookeeper.admin.forceHTTPS", "true");
+        System.setProperty("zookeeper.admin.portUnification", String.valueOf(portUnification));
+        boolean httpsPassed = false;
+
+        JettyAdminServer server = new JettyAdminServer();
+        try {
+            server.start();
+            queryAdminServer(String.format(HTTPS_URL_FORMAT, jettyAdminPort), true);
+            httpsPassed = true;
+            queryAdminServer(String.format(URL_FORMAT, jettyAdminPort), false);
```

Comment:
I think currently the test will pass always, even if the http call succeeds.

I think you should add a line here (after the http call), like:
`fail("http call should have been failed when forceHTTPS=true");` 
to actually assert that the http call fails, the https is enforced and we jumped to the `catch` block.

Reply:
Thanks, true that, I had that fail but got lost after the refactor! Good catch (I even had a spotbugs failure of unused import of Assert.fail, and I didn't realize my mistake, removed the import instead. duh...)